gateway/api_test.go

```diff
@@ -1528,7 +1532,14 @@ func TestGroupResetHandler(t *testing.T) {
 	// We wait for the right notification (NoticeGroupReload), other
 	// type of notifications may be received during tests, as this
 	// is the cluster channel:
-	<-didReload
+	select {
+	case <-time.After(time.Second):
```

Comment:
I know this are tests, but this will introduce memory leak, when `<-didReload` happens function will exit and `<-time.After` resources will not get garbage collected.

Reply:
We're going to leave this to fix in tests code in a later sprint, when we'll be addressing flaky tests and other concerns. Thanks for the heads up - I did fix the bit of code that wasn't in a _test.go file and had the same pattern (analytics records flusher)